README.md

```diff
@@ -1,2 +1,124 @@
+You should be able to use either version 1 (>v1.17) or version 2 of the AWS command line tools, but these instructions have been primarily tested with v1.18.
+To install, you can follow [amazon's instructions for v2](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) or install via your favorite package manager (`pip` and `conda` will at this time install v1, `homebrew` will install v2):
+- `pip install awscli`
+- `conda install -c conda-forge awscli`
+- `brew install awscli`
+
+#### Configuring your AWS credentials
+
+When you set up your account, you may have gotten a paired `AWS Access Key ID` and `AWS Secret Access Key`.
+If you did not, you can login to the [aws console](https://console.aws.amazon.com/), select "My Security Credentials" from the menu that appears clicking on your username in the upper right.
+Scroll down to "Access keys for CLI, SDK, & API access" and click the "Create Access Key" button.
+You will see a popup with your Access key ID and a link to show the Secret access key.
+
+In the terminal, run the command `aws configure` and when prompted paste in the `AWS Access Key ID` and `AWS Secret Access Key`.
+You may want to set up your `Default region name`: most of the computing resources used here run in `us-east-1`, but setting this to a different region should not affect most commands.
+The `Default output format` can be left as `None`
+
+The `aws configure` command will create a directory at `~/.aws` with a `config` file and a `credentials` file that will be required for `nextflow` commands to work smoothly with Batch and S3.
+
+#### AWS infrastructure
+
+The infrastructure on AWS (Batch queues, AMIs, etc.) is defined via [terraform](https://www.terraform.io) using the files in the `aws` directory. More details are described in [aws/setup-log.md](aws/setup-log.md)
+
+
+
+## Running Workflows
+
+The workflows for this repository are stored in the `workflows` directory, organized by task into subdirectories.
+In general, workflows should be run from the main `workflows` directory, rather than the subdirectories.
+This will allow easy access to the shared `nextflow.config` file and keep all intermediate and log files in a single location.
+In particular, `nextflow` creates a separate work directory for every task: these will appear by default (for local tasks) in the `workflows/work` directory if the command is invoked from `workflows`.
+As the work directories can get large, it is helpful to have a single location to keep track of and purge as needed.
+
+## Running locally
+
+⚠️ Running workflows in this repository locally is not something to take lightly! It is fine for a quick test, and useful for development, but note that most of the workflows here use very large data files, which will have to be downloaded locally to run. Workflow processing may require large amounts of RAM and time, so the following example commands will rarely be used, and are mostly for illustration. ⚠️
+
```

Comment:
Nextflow told me: 
`Caused by:
  Process requirement exceed available memory -- req: 28 GB; avail: 16 GB` I don't know how it knows 28 GB is what it needs, but you could put that down somewhere (if that is accurate).

Reply:
hmm.. yeah, that's what I get for not testing instructions. I could give an example that will run locally, but all the mapping workflows request more processors... I think that is something I should fix at the config level. I want to have a local example, as I use them often for testing, and I wanted to build up the important nextflow options sequentially.

Reply:
Okay, fixed the workflow to only request lots of memory and cpu if running in batch. It should work now, restoring the state I meant to have where running locally deserved a caution, but should not outright fail (except cellranger, which will fail because it is cellranger).

Reply:
I'll try running again then.